core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariClusterConnector.java

```diff
@@ -125,7 +127,8 @@ public Cluster buildAmbariCluster(Stack stack) {
             AmbariClient ambariClient = ambariClientProvider.getAmbariClient(cluster.getAmbariIp(), cluster.getUserName(), cluster.getPassword());
             setBaseRepoURL(cluster, ambariClient);
             addBlueprint(stack, ambariClient, cluster.getBlueprint());
-            PollingResult waitForHostsResult = waitForHosts(stack, ambariClient);
+            PollingResult waitForHostsResult = waitForHosts(stack, ambariClient,
+                    stack.getFullNodeCountWithoutDecommissionedNodes() - stack.getGateWayNodeCount());
```

Comment:
I would use an individual variable for calculating the required number of nodes.